Dockerfile.cuda

```diff
@@ -4,15 +4,18 @@ ARG TARGETARCH
 ARG VERSION=0.0.0
 
 WORKDIR /go/src/github.com/jmorganca/ollama
-RUN apt-get update && apt-get install -y git build-essential cmake
+RUN apt-get update && apt-get install -y git build-essential cmake nvidia-cuda-toolkit
 ADD https://dl.google.com/go/go1.21.1.linux-$TARGETARCH.tar.gz /tmp/go1.21.1.tar.gz
 RUN mkdir -p /usr/local && tar xz -C /usr/local </tmp/go1.21.1.tar.gz
 
 COPY . .
 ENV GOARCH=$TARGETARCH
-RUN /usr/local/go/bin/go generate ./... \
+RUN CUDA_VERSION=$(nvcc --version | sed -n 's/^.*release \([0-9]\+\)\.\([0-9]\+\).*$/\1/p') && \
+    /usr/local/go/bin/go generate ./... \
     && /usr/local/go/bin/go build -ldflags "-linkmode=external -extldflags='-static' -X=github.com/jmorganca/ollama/version.Version=$VERSION -X=github.com/jmorganca/ollama/server.mode=release" .
```

Comment:
This is a bug right now where `CUDA_VERSION` _is_ set by the base image but it's set in a way that the runner selection doesn't see it. However @BruceMacD is currently working on removing need for this var